books/projects/x86isa/utils/utilities.lisp

```diff
@@ -507,29 +558,47 @@ constants and functions; it also proves some associated lemmas.</p>"
         ;; XY-bit integer recognizer
         (signed-byte-p ,n x))
 
+     `(define ,iXY ((x integerp))
+        ;; Truncate input to an XY-bit signed integer number
+        ;; This function can be used to convert, say a 32-bit natural number
+        ;; to a 32-bit integer.  We choose to keep this function
+        ;; enabled.
+        :inline t
+        :enabled t
+        :parents (constants-conversions-and-bounds)
+        (logext ,n x))
+
      `(define ,ntoi
         :inline t
         :enabled t
         :parents (constants-conversions-and-bounds)
         ;; Convert natural number to integer
-        :guard-hints (("Goal" :in-theory (e/d (logext)
-                                              (unsigned-byte-p))))
+        :guard-hints (("Goal" :in-theory (enable logext)))
         ((x ,nXYp :type (unsigned-byte ,n)))
 
         (mbe :logic (logext ,n x)
              :exec (if (< x ,(expt 2 (1- n)))
                        x
-                     (- x ,(expt 2 n))))
-
-        ///
+                     (- x ,(expt 2 n)))))
 
-        (defthm-sb ,(mk-name iXYp "-" ntoi)
```

Comment:
Regressions pass without this theorem because the defintion `iWp` is enabled by default. Remove this theorem for faster proofs.